tycho-common/examples/chain/mod.rs

```diff
@@ -0,0 +1,41 @@
+//! Action chaining system for composing multi-step DeFi operations.
+//!
+//! This module provides a type-safe builder pattern for creating chains of actions
```

Comment:
I find that using the name "Chain" for this might be very confusing.. chain already means something in this context 😬 maybe sequence?